cmd/go-clang-generate/struct.go

```diff
@@ -17,218 +16,98 @@ type Struct struct {
 
-				callToC := receiver + ".c." + cursor.DisplayName()
+		switch cursor.Kind() {
+		case clang.CK_FieldDecl:
+			conv := getTypeConversion(cursor.Type())
 
-				var method string
-
-				if conv.Pointer == 2 {
-					elemType := "&" + f.ReturnType
-					f.ReturnType = "[]*" + f.ReturnType
+			if conv.IsFunctionPointer {
+				return clang.CVR_Continue
+			}
 
-					var b bytes.Buffer
-					if err := templateGenerateReturnSlice.Execute(&b, f); err != nil {
-						fmt.Println(err.Error())
-					}
+			comment := cleanDoxygenComment(cursor.RawCommentText())
 
-					method = fmt.Sprintf(b.String(), callToC, callToC, elemType, callToC)
-				} else if conv.Pointer < 2 {
-					if conv.Pointer == 1 {
-						f.ReturnType = "*" + f.ReturnType
-					}
+			var method string
 
-					if f.ReturnType == "*void" {
-						f.ReturnType = GoPointer
-						s.ImportUnsafe = true
-					}
+			if conv.PointerLevel == 2 {
 
-					if conv.IsArray {
-						f.ReturnType = "[]" + f.ReturnType
-					}
+				f := &FunctionSliceReturn{
+					Function: *generateFunction(cursor.DisplayName(), cname, comment, cursor.DisplayName(), conv),
 
-					if conv.Primitive {
-						f.ReturnString = fmt.Sprintf(returnPrimitive, strings.Replace(f.ReturnType, "*", "&", -1), callToC)
-					} else {
-						f.ReturnString = fmt.Sprintf(returnComplex, strings.Replace(f.ReturnType, "*", "&", -1), callToC)
-					}
+					ElementType:          "*" + conv.GoType,
+					IsElementTypePointer: true,
+					PointeeType:          conv.GoType,
+					IsPrimitive:          conv.IsPrimitive,
+				}
 
-					var b bytes.Buffer
-					if err := templateGenerateGetter.Execute(&b, f); err != nil {
-						fmt.Println(err.Error())
-					}
+				method = generateFunctionSliceReturn(f)
+				s.ImportStdLib = true
 
-					method = b.String()
-				} else {
-					panic("Three pointers")
+			} else if conv.PointerLevel < 2 {
+				if conv.PointerLevel == 1 && conv.GoType == "void" {
+					conv.GoType = GoPointer
+					conv.PointerLevel = 0
+					s.ImportUnsafe = true
 				}
 
-				s.Methods = append(s.Methods, method)
+				f := generateFunction(cursor.DisplayName(), cname, comment, cursor.DisplayName(), conv)
+
+				method = generateFunctionStructMemberGetter(f)
+			} else {
+				panic("Three pointers")
 			}
 
-			return clang.CVR_Continue
-		})
-	}
+			s.Methods = append(s.Methods, method)
+		}
+
+		return clang.CVR_Continue
+	})
 
 	return s
 }
 
-const (
-	GoInt8      = "int8"
-	GoUInt8     = "uint8"
-	GoInt16     = "int16"
-	GoUInt16    = "uint16"
-	GoInt32     = "int32"
-	GoUInt32    = "uint32"
-	GoInt64     = "int64"
-	GoUInt64    = "uint64"
-	GoFloat32   = "float32"
-	GoFloat64   = "float64"
-	GoBool      = "bool"
-	GoInterface = "interface"
-	GoPointer   = "unsafe.Pointer"
-)
+func generateFunction(name, cname, comment, member string, conv Conversion) *Function {
+	receiverType := trimClangPrefix(cname)
+	receiverName := strings.ToLower(string(receiverType[0]))
```

Comment:
Use `receiverName(receiverType[0])` here. It will generate a "nice" receiver name.